submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/api/FileStorageService.kt

```diff
@@ -11,5 +11,9 @@ interface FileStorageService {
 
     fun deleteSubmissionFile(sub: ExtSubmission, file: ExtFile)
 
-    fun deleteSubmissionFiles(sub: ExtSubmission)
+    fun deleteSubmissionFiles(
+        sub: ExtSubmission,
+        onEachIndexed: (Int, ExtFile) -> Unit = { _, _ -> },
```

Comment:
I understand the idea behind this parameter but I'm not sure we should have it since I think it makes the signature and the code more complicated than it should be. I think that since the operation is specifically for deleting and we're currently only using this parameter for logging, we'd be better off leaving the logging to the implementation of this method so it would be just

```
fun deleteSubmissionFiles(sub: ExtSubmission, filter: (ExtFile) -> Boolean = { true })
```

(The comment continues below 1/3)